### PR TITLE
Feature/application m2 m users

### PR DESCRIPTION
**Pull request by GrahamMWillis**
Migrate to the following relationships

User <=> Application
User <=> Site
Application <=> Site

Make changes to the 

- OpenAPISpecifications
- Database
- API Handlers
- Application Queue
- Application Extracts
- Front End

See https://eaflood.atlassian.net/browse/SDDSIP-357

### CODE REVIEW

File: packages/api/curl-scripts/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+For testing
```
Comment (kealanparr) [inline packages/api/curl-scripts/README.md:1]: Do you think it might be worth adding some example cURL requests?

Comment (kealanparr) [reply packages/api/curl-scripts/README.md:1]: I think you could even just point to `packages/api/curl-scripts/setup-applications.sh` as that has them!

File: packages/api/openapi/licence.yaml [hunk 2]
```diff
@@ -164,18 +164,24 @@ paths:
               schema:
                 $ref: '#/components/schemas/error'
 
-  '/user/{userId}/applications':
+  '/applications':
```
Comment (kealanparr) [inline packages/api/openapi/licence.yaml:167]: Is this file auto-generated?

Comment (GrahamMWillis) [reply packages/api/openapi/licence.yaml:167]: No!

File: packages/api/curl-scripts/README.md (hunk 1 above)
Comment (GrahamMWillis) [reply packages/api/curl-scripts/README.md:1]: Its documented in the swagger-ui really and you can generate curl request from that.

File: packages/api/openapi/licence.yaml (hunk 2 above)
Comment (kealanparr) [reply packages/api/openapi/licence.yaml:167]: Oh my life 😂 This must have been like 99% of the time taken haha!

File: packages/web-service/src/constants.js [hunk 3]
```diff
@@ -1,2 +1,3 @@
 export const SESSION_TTL_MS_DEFAULT = 3 * 60 * 60 * 1000
 export const SESSION_COOKIE_NAME_DEFAULT = 'sid'
+export const DEFAULT_ROLE = 'USER' // Will be replaced
```
Comment (kealanparr) [inline packages/web-service/src/constants.js:3]: Is this coming in a follow up PR? Or fixed much further down the line?

File: packages/api/src/handlers/application-site/__tests__/application-site.spec.js [hunk 4]
```diff
@@ -93,8 +88,7 @@ describe('The postApplicationSite handler', () => {
     expect(codeFunc).toHaveBeenCalledWith(201)
   })
 
-  it('returns a conflict and error if the user-application-site already exists', async () => {
-    models.users = { findByPk: jest.fn(async () => ({ dataValues: { id: 'bar' } })) }
+  it('returns a conflict and error if the application-site already exists', async () => {
```
Comment (kealanparr) [inline packages/api/src/handlers/application-site/__tests__/application-site.spec.js:91]: Couldn't we have multiple applications on one site?

File: packages/api/curl-scripts/README.md (hunk 1 above)
Comment (kealanparr) [reply packages/api/curl-scripts/README.md:1]: Fair enough, will resolve then

File: packages/web-service/src/constants.js (hunk 3 above)
Comment (GrahamMWillis) [reply packages/web-service/src/constants.js:3]: It's further down the line. We have to determine how the roles will work, we may have things like APPROVER, CREATOR, SUBMITTER, READ-ONLY etc.

File: packages/api/src/handlers/application-site/__tests__/application-site.spec.js (hunk 4 above)
Comment (GrahamMWillis) [reply packages/api/src/handlers/application-site/__tests__/application-site.spec.js:91]: Yes indeed we can. The relation will allow it. Each site-application relation is unique though.

**Pull request merged**